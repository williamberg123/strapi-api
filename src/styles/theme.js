const theme = {
	colors: {
		mainBg: 'red',
		secondaryBg: 'blue'
	},
	fonts: {
		small: '0.75rem',
		normal: '1rem',
		medium: '1.5rem',
		big: '2rem'
	},
	spacings: {},
};

export default theme;
