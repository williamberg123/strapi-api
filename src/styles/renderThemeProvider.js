import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from './theme';

const renderThemeProvider = (children) => {
	return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

export default renderThemeProvider;
