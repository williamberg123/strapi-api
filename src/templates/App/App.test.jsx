import { screen } from '@testing-library/react';

import renderThemeProvider from '../../styles/renderThemeProvider';
import Home from '.';
import theme from '../../styles/theme';

it('should render Home correctly', () => {
	renderThemeProvider(<Home />);
	const headingContainer = screen.getByRole('heading', { name: /oi/i }).parentElement;
	expect(headingContainer).toBeInTheDocument();
	expect(headingContainer).toMatchSnapshot();
});
