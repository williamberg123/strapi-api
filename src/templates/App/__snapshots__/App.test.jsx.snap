// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home correctly 1`] = `
.c0 {
  width: 100%;
}

<div
  class="c0"
>
  <h1>
    Oi
  </h1>
</div>
`;
